platform/src/FileHandler.js

```diff
@@ -101,12 +125,35 @@ export class FileHandler {
 
             // At this point, this is either a public repository, or it's a private repository but an unknown type of URL.
             // In either case, we assume that we can simply access the URL directly.
-            const responseText = await utility.getRequest(url, false);
-            return { content: responseText, sha: null };
+            const response = await this.refactoredFetchFileDirectly(url);
+            if (response != null) {
+                return { content: response.content, sha: null }; // TODO need to retrieve the sha for the file
+            }
+            return null;
+
+        } 
+        catch (error) {
+            console.error("Failed to fetch file with url: " + url, error);
+            return null;
+        }
+    }
 
+    /**
+     * TODO: Use this instead of the above
+     * Rename this to fetchFileDirectly and remove the above function.
+     * 
+     * Fetch file directly via HTTP without using VCS providers (async version).
+     * Used for tool-generated content and other non-repository URLs.
+     * @param {String} url - The direct URL to fetch
+     * @returns {Promise<{content: string} | null>} Promise resolving to object with content, or null if failed
+     */
+    async refactoredFetchFileDirectly(url) {
+        try {
+            const responseText = await utility.getRequest(url, false);
+            return { content: responseText };
         } 
         catch (error) {
-            console.error("Failed to fetch file:", error);
+            console.error("Failed to fetch file with url: " + url, error);
```

Comment:
Should this be allowed to float through to the platform to be shown as an error to the user?